src/components/Toolbar/StatusFilter/index.jsx

```diff
@@ -0,0 +1,67 @@
+import { useState, useEffect } from "react";
+
+function StatusFilter({ setFilterStatus }) {
+  const [status, setStatus] = useState();
+  const [loading, setLoading] = useState(true);
+
+  const INVOICE_URL = "https://invoicing-api.dev.io-academy.uk/invoices";
+
+  const fetchStatus = async () => {
+    const res = await fetch(INVOICE_URL);
+    const data = await res.json();
+    setStatus(data);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    fetchStatus();
+  }, []);
+
+  let completeStatus = [];
+
+  if (!loading) {
+    let newStatusArray = [];
+    status.data.map((item) => {
+      newStatusArray.push({ status: item.status, name: item.status_name });
+    });
+
+    let statusNumber = [];
+
+    newStatusArray.map((item) => {
+      if (statusNumber.includes(item.status) === false) {
+        statusNumber.push(item.status);
+      }
+    });
+
+    let statusName = [];
+    newStatusArray.map((item) => {
+      if (statusName.includes(item.name) === false) {
+        statusName.push(item.name);
+      }
+    });
```

Comment:
Same goes for these. Pushing to an array, and mutating it, isn't very functional in style!